packages/flutter/lib/src/cupertino/dialog.dart

```diff
@@ -1128,9 +1135,47 @@ class _CupertinoActionSheetActionState extends State<CupertinoActionSheetAction>
     widget.onPressed();
   }
 
+  // Calculates the font size for action sheet buttons, which deviate from
+  // standard HIG specifications.
+  //
+  // There is a non-linear relationship between the font size for regular body
+  // text and the font size for the action sheet buttons:
+  //
+  //  Text scale  | xs |  s |  m |  l | xl | xxl | xxxl | ax1 | ax2 | ax3 | ax4 | ax5
+  //  Body font   | 14 | 15 | 16 | 17 | 19 |  21 |  23  |  28 |  33 |  40 |  47 |  53
+  //  Button font | 21 | 21 | 21 | 21 | 23 |  24 |  24  |  28 |  33 |  40 |  47 |  53
+  //
+  // The `contextBodySize` is the body font size specified by context. The
+  // return value is the button font size, including the effect of context font
+  // scale factor. Divide by context font scale factor before using in a `Text`.
+  static double _buttonFontSize(double contextBodySize) {
+    // For very small or very large text, simpler scaling mappings are used.
+    if (contextBodySize <= 17) {
+      return 21;
+    }
+    if (contextBodySize >= 24.5) {
+      return contextBodySize;
+    }
+    // For mid-sized text, an interpolated curve is used. The breakpoint for
+    // switching between these behaviors is 24.5 instead of 28 to allow closer
+    // interpolation over a smaller range.
+    final double resultSize = -18.8 + 3.68 * contextBodySize * (1 - 0.02128 * contextBodySize);
+    return resultSize.roundToDouble();
+  }
+
   @override
   Widget build(BuildContext context) {
+    // The context scale factor is derived from the current body size and the
+    // standard body size in "large".
+    const double higLargeBodySize = 17.0;
+    final double contextBodySize = MediaQuery.textScalerOf(context).scale(higLargeBodySize);
+    final double contextScaleFactor = contextBodySize / higLargeBodySize;
+    final double fontSize = _buttonFontSize(contextBodySize);
```

Comment:
ah so the issue is that buttonfont size doe not scale linear with contextBodySize? that is unfortunate...